hardware/esp8266com/esp8266/libraries/ESP8266WiFi/src/WiFiClient.cpp

```diff
@@ -105,11 +110,14 @@ int ICACHE_FLASH_ATTR  WiFiClient::connect(IPAddress ip, uint16_t port)
             g_localPort = MIN_LOCAL_PORT;
         if (err == ERR_OK)
             break;
-        if (err == ERR_USE)
+        if (err == ERR_USE) {
+        	esp_yield();
```

Comment:
we are here in the sketch task. it is the connect function of the class.
some note the code is disabled currently (the NO_PORT_BIND define is set)